packages/core-js/utils/ast.js

```diff
@@ -0,0 +1,99 @@
+const { findAll } = require('solidity-ast/utils');
+
+function findContractNodeWithId(contractId, asts) {
+  return Array.from(findAll('ContractDefinition', asts)).find(
+    (contractDefiniton) => contractDefiniton.id === contractId
+  );
+}
+
+function findContractNodeWithName(contractName, asts) {
+  return Array.from(findAll('ContractDefinition', asts)).find(
+    (contractDefiniton) => contractDefiniton.name === contractName
+  );
+}
+
+function getContractNode(ast) {
+  return Array.from(findAll('ContractDefinition', ast))[0];
+}
+
+function findContractNodeVariables(contractNode) {
+  return Array.from(findAll('VariableDeclaration', contractNode));
+}
+
+function findContractStateVariables(contractName, ast) {
+  return findContractNodeVariables(getContractNode(ast)).filter((n) => n.stateVariable);
+}
+
+function findContractDependencies(contractName, asts) {
+  const contractNode = getContractNode(asts[contractName]);
+
+  let dependencyContractNodes = [];
+
+  contractNode.linearizedBaseContracts.forEach((baseContractId) => {
+    const dependency = findContractNodeWithId(baseContractId, asts);
+    if (dependency) {
+      dependencyContractNodes.push(dependency);
+    }
+  });
+
+  return dependencyContractNodes;
```

Comment:
np: use same return format for all array related on all utils. 
i.e. 
```
let dependencyContractNodes = [];
...
return dependencyContractNodes;

```
or 
```
const slots = [];
...
return slots ? slots : null;

```
(I prefer returning always the array, empty if no results found.)

Reply:
Done! All functions now return the array and not null.